### ISSUE TITLE
Strange bicycle route

### ISSUE THREAD

**Issue by starsep** [organicmaps/organicmaps#4059]
Calculated route is longer, has 2 extra traffic signals and bridge.
https://osm.org/directions?engine=fossgis_osrm_bike&route=52.15125%2C21.09130%3B52.15546%2C21.08956

| Organic Maps | OSRM |
| - | - |
| ![image](https://user-images.githubusercontent.com/2798728/206771896-7642cf93-bcbd-4694-9178-beb0e1627984.png) | ![image](https://user-images.githubusercontent.com/2798728/206771397-402ef475-f76d-4767-934b-ad46281e6832.png) |

**Comment by vng** [comment]
Yes, app prefers pure highway=cycleway roads better than say highway=path + bicycle=yes.
I a draft solution for traffic_signals penalty, but now it's turned-off because car routing is fancy sometimes (need to add also turns penalty). I think, we can enable traffic_signals for pedestrian/bicycle crossings.

**Comment by mangerlahn** [comment]
I may have a similar case, not as obvious though. Should I add this as a separate issue?
https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=51.33772%2C12.41432%3B51.33837%2C12.40958#map=18/51.33805/12.41217&layers=Y

<img width="1071" alt="Screenshot 2023-01-21 at 22 43 32" src="https://user-images.githubusercontent.com/11056766/213888161-0abddb74-3595-43c7-ba07-5fdd45f8c44d.png">
<img width="1071" alt="Screenshot 2023-01-21 at 22 43 32" src="https://user-images.githubusercontent.com/11056766/213888174-e93e5f6d-2a8d-429e-b74e-2bc72a310b10.PNG">

**Comment by Jean-BaptisteC** [comment]
First case is fixed

**Comment by pastk** [comment]
> https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=51.33772%2C12.41432%3B51.33837%2C12.40958#map=18/51.33805/12.41217&layers=Y

This case is even worse now
![image](https://github.com/user-attachments/assets/3b8751ab-46c3-49e4-99ca-c36182b25950)

`bicycle=yes` has too much preference, see https://github.com/organicmaps/organicmaps/issues/1772
